bob-common/src/mapper.rs

```diff
@@ -121,6 +121,44 @@ impl Virtual {
         self.vdisks.get(&id).expect("vdisk not found").nodes()
     }
 
+    fn find_index(&self, offset: usize) -> Offset {
+        let mut avail = 0;
+        for i in 0..self.nodes.len() {
+            let node = &self.nodes[i];
+            if node.connection_available() {
```

Comment:
You forget about the following condition: `target_nodes.iter().all(|n| n.index() != node.index())`